### PR TITLE
Fix libxslt to build on Android

### PR DESCRIPTION
**Pull request by dmn-star**
Specify library name and version:  **libxslt/all**

Fixes: #1610
Fixes: #2181
 
Wrong libxml2 include/lib detection on MacOS fixed too
see https://github.com/conan-io/conan-center-index/issues/1610#issuecomment-629308437

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by prince-chrismc** [comment]
How does this address the `pthead` issue? is not obvious from the diff

**Comment by dmn-star** [comment]
Yes, the configure going crazy

LIBXML_CFLAGS='-I/Library/Developer/CommandLineTools/SDKs/MacOSX10.14.sdk/usr/include/libxml2'
LIBXML_LIBS='-lxml2 -lz -**lpthread** -**licucore** -lm'

**Review by SSE4 — APPROVED** [review]

File: recipes/libxslt/all/conanfile.py [hunk 1]
```diff
@@ -125,9 +125,9 @@ def _build_with_configure(self):
         else:
             configure_args.extend(['--enable-static', '--disable-shared'])
 
-        xml_config = tools.unix_path(self.deps_cpp_info["libxml2"].rootpath) + "/bin/xml2-config"
-
-        configure_args.append('XML_CONFIG=%s' % xml_config)
+        libxml2_path = tools.unix_path(self.deps_cpp_info["libxml2"].rootpath)
+        libxml_src = ("--with-libxml-src=%s" %libxml2_path)
```
Comment (theirix) [inline recipes/libxslt/all/conanfile.py:129]: Just for better readability
```suggestion
        libxml_src = "--with-libxml-src=%s" % libxml2_path
```

**Review by theirix — CHANGES_REQUESTED** [review]

File: recipes/libxslt/all/conanfile.py (hunk 1 above)
Comment (dmn-star) [reply recipes/libxslt/all/conanfile.py:129]: May simple libxml_src = "--with-libxml-src=" + tools.unix_path(self.deps_cpp_info["libxml2"].rootpath)

What do you think?

Comment (dmn-star) [reply recipes/libxslt/all/conanfile.py]: Done

**Review by prince-chrismc — APPROVED** [review]

**Review by madebr — APPROVED** [review]

**Review by SSE4 — APPROVED** [review]

**Pull request merged**